modules/imgproc/src/intersection.cpp

```diff
@@ -114,10 +122,20 @@ static int _rotatedRectangleIntersection( const RotatedRect& rect1, const Rotate
             float vx2 = vec2[j].x;
             float vy2 = vec2[j].y;
 
+            float numericalScalingFactor = std::min(std::abs(vx1*vx1+vy1*vy1), std::abs(vx2*vx2+vy2*vy2));
```

Comment:
Why is `std::min`?
We need to "eliminate" large numbers.

Reply:
I think both min and max are valid and (more or less) equivalent (relatively to numerical stability) for this use case.
 The idea is to scale the range to be close to 1, so either you do it relatively to the lower bound (with a min), or to the higher bound (with a max); we could even scale relatively to the mean value